File: frontend/scripts/parse-wordpress-xml.js
const fs = require('fs');
const { parseString } = require('xml2js');
const path = require('path');

// Read the XML file
const xmlFile = path.join(__dirname, '../../stakquedigital.WordPress.2025-12-03-2.xml');
const xmlContent = fs.readFileSync(xmlFile, 'utf-8');

parseString(xmlContent, (err, result) => {
  if (err) {
    console.error('Error parsing XML:', err);
    return;
  }

  const items = result.rss.channel[0].item || [];
  const posts = [];

  items.forEach((item, index) => {
    const title = item.title?.[0]?._ || item.title?.[0] || '';
    const link = item.link?.[0] || '';
    const pubDate = item.pubDate?.[0] || '';
    const content = item['content:encoded']?.[0]?._ || item['content:encoded']?.[0] || '';
    const description = item.description?.[0]?._ || item.description?.[0] || '';
    const creator = item['dc:creator']?.[0]?._ || item['dc:creator']?.[0] || '';
    
    // Extract category
    const categories = item.category || [];
    const category = categories.find(cat => cat.$.domain === 'category')?._ || 
                     categories.find(cat => cat.$.domain === 'post_tag')?._ || 
                     categories[0]?._ || '';

    // Generate slug from title
    const slug = title
      .toLowerCase()
      .replace(/[^a-z0-9]+/g, '-')
      .replace(/^-+|-+$/g, '')
      .substring(0, 100);

    // Extract excerpt from content (first 200 chars)
    const excerpt = description || 
                   content.replace(/<[^>]*>/g, '').substring(0, 200).trim() + '...';

    // Parse date
    let publishedAt = new Date().toISOString();
    if (pubDate) {
      const date = new Date(pubDate);
      if (!isNaN(date.getTime())) {
        publishedAt = date.toISOString();
      }
    }

    // Only include posts with content
    if (title && content && slug) {
      posts.push({
        id: index + 1,
        title,
        slug,
        excerpt,
        content,
        category: category || undefined,
        publishedAt,
        author: creator || 'Stakque Team'
      });
    }
  });

  // Sort by date (newest first)
  posts.sort((a, b) => new Date(b.publishedAt) - new Date(a.publishedAt));

  // Generate TypeScript file
  const tsContent = `// Auto-generated from WordPress XML export
// Generated on ${new Date().toISOString()}

export interface MockBlogPost {
  id: number;
  title: string;
  slug: string;
  excerpt?: string;
  content: string;
  category?: string;
  publishedAt: string;
  author?: string;
}

export const mockBlogPosts: MockBlogPost[] = ${JSON.stringify(posts, null, 2)};

export default mockBlogPosts;
`;

  const outputFile = path.join(__dirname, '../lib/mockBlogPosts.ts');
  fs.writeFileSync(outputFile, tsContent, 'utf-8');
  
  console.log(`✅ Parsed ${posts.length} blog posts`);
  console.log(`📝 Generated ${outputFile}`);
  console.log(`\nFirst 5 posts:`);
  posts.slice(0, 5).forEach((post, i) => {
    console.log(`${i + 1}. ${post.title} (${post.slug})`);
  });
});

File: frontend/scripts/parse_wordpress_xml.py
#!/usr/bin/env python3
import xml.etree.ElementTree as ET
import re
import json
from datetime import datetime
from html import unescape

# Read the XML file
xml_file = '../../stakquedigital.WordPress.2025-12-03-2.xml'
tree = ET.parse(xml_file)
root = tree.getroot()

# Define namespaces
namespaces = {
    'content': 'http://purl.org/rss/1.0/modules/content/',
    'wp': 'http://wordpress.org/export/1.2/',
    'dc': 'http://purl.org/dc/elements/1.1/'
}

posts = []
items = root.findall('.//item')

for index, item in enumerate(items):
    title_elem = item.find('title')
    link_elem = item.find('link')
    pub_date_elem = item.find('pubDate')
    content_elem = item.find('content:encoded', namespaces)
    description_elem = item.find('description')
    creator_elem = item.find('dc:creator', namespaces)
    
    # Get categories
    categories = item.findall('category')
    category = None
    for cat in categories:
        if cat.get('domain') == 'category' or cat.get('domain') == 'post_tag':
            category = cat.text
            break
    if not category and categories:
        category = categories[0].text
    
    if title_elem is not None and content_elem is not None:
        title = unescape(title_elem.text or '').strip()
        content = unescape(content_elem.text or '').strip()
        link = link_elem.text if link_elem is not None else ''
        pub_date = pub_date_elem.text if pub_date_elem is not None else ''
        description = unescape(description_elem.text or '').strip() if description_elem is not None else ''
        creator = creator_elem.text if creator_elem is not None else 'Stakque Team'
        
        # Generate slug from title
        slug = re.sub(r'[^a-z0-9]+', '-', title.lower()).strip('-')[:100]
        
        # Extract excerpt
        excerpt = description
        if not excerpt:
            # Remove HTML tags and get first 200 chars
            text_content = re.sub(r'<[^>]+>', '', content)
            excerpt = text_content[:200].strip() + '...' if len(text_content) > 200 else text_content
        
        # Parse date
        published_at = datetime.now().isoformat()
        if pub_date:
            try:
                # Parse RFC 2822 date format
                from email.utils import parsedate_to_datetime
                dt = parsedate_to_datetime(pub_date)
                published_at = dt.isoformat()
            except:
                pass
        
        if title and content and slug:
            posts.append({
                'id': index + 1,
                'title': title,
                'slug': slug,
                'excerpt': excerpt,
                'content': content,
                'category': category,
                'publishedAt': published_at,
                'author': creator
            })

# Sort by date (newest first)
posts.sort(key=lambda x: x['publishedAt'], reverse=True)

# Generate TypeScript file
ts_content = f"""// Auto-generated from WordPress XML export
// Generated on {datetime.now().isoformat()}
// Total posts: {len(posts)}

export interface MockBlogPost {{
  id: number;
  title: string;
  slug: string;
  excerpt?: string;
  content: string;
  category?: string;
  publishedAt: string;
  author?: string;
}}

export const mockBlogPosts: MockBlogPost[] = {json.dumps(posts, indent=2, ensure_ascii=False)};

export default mockBlogPosts;
"""

output_file = '../lib/mockBlogPosts.ts'
with open(output_file, 'w', encoding='utf-8') as f:
    f.write(ts_content)

print(f'✅ Parsed {len(posts)} blog posts')
print(f'📝 Generated {output_file}')
print(f'\nFirst 5 posts:')
for i, post in enumerate(posts[:5]):
    print(f'{i+1}. {post["title"][:60]}... ({post["slug"]})')

